wrangler-service/src/main/java/io/cdap/wrangler/service/directive/DirectivesHandler.java

```diff
@@ -816,6 +823,173 @@ public void schema(HttpServiceRequest request, HttpServiceResponder responder,
     });
   }
 
+  /**
+   * Adds a data model to workspace. There is a one-to-one mapping between workspaces and models.
+   *
+   * @param request Handler for incoming request.
+   * @param responder Responder for data going out.
+   * @param id Workspace to associate a data model.
+   */
+  @POST
+  @Path("contexts/{context}/workspaces/{id}/datamodels")
```

Comment:
why does this need to be in directives handler? can this be moved to datamodel handler?

Reply:
This end point is updating properties of the workspace that will be used to maintain workspace related state client side. Also, following a similar pattern to the schema route within directives handler.